src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/CannotAddAbstractMemberTests.cs

```diff
@@ -0,0 +1,353 @@
+            ElementContainer<IAssemblySymbol> left =
+                new(SymbolFactory.GetAssemblyFromSyntax(leftSyntax), new MetadataInformation(string.Empty, string.Empty, "ref"));
+
+            IList<ElementContainer<IAssemblySymbol>> right = SymbolFactory.GetElementContainersFromSyntaxes(rightSyntaxes);
+
+            IEnumerable<(MetadataInformation, MetadataInformation, IEnumerable<CompatDifference>)> differences =
+                differ.GetDifferences(left, right);
+
+            CompatDifference[][] expectedDiffs =
+            {
+                Array.Empty<CompatDifference>(),
+                new[]
+                {
+                    new CompatDifference(DiagnosticIds.CannotAddAbstractMember, string.Empty, DifferenceType.Added, "M:CompatTests.First.FirstNested.SecondNested.SomeAbstractMethod"),
+                },
+                new[]
+                {
+                    new CompatDifference(DiagnosticIds.CannotAddAbstractMember, string.Empty, DifferenceType.Added, "M:CompatTests.First.FirstNested.FirstNestedAbstract"),
+                },
+            };
+
+            AssertExtensions.MultiRightResult(left.MetadataInformation, expectedDiffs, differences);
+        }
+
+        public static IEnumerable<object[]> AddedToUnsealedTypeInRightNotReportedData()
+        {
+            yield return new object[]
+            {
+                @"
+namespace CompatTests
+{
+  public sealed class First
+  {
+    public void SomeMethod() { }
+  }
+}
+",
+                @"
+namespace CompatTests
+{
+  public class First
+  {
+    public void SomeMethod() { }
+    public abstract void SomeAbstractMember() { }
+  }
+}
+"
+            };
+
+            yield return new object[]
+            {
+                @"
+namespace CompatTests
+{
+  public class First
+  {
+    private First() { }
+    public void SomeMethod() { }
+  }
+}
+",
+                @"
+namespace CompatTests
+{
+  public class First
+  {
+    public First() { }
+    public void SomeMethod() { }
+    public abstract void SomeAbstractMember() { }
+  }
+}
+"
+            };
+
+            yield return new object[]
+            {
+                @"
+namespace CompatTests
+{
+  public class First
+  {
+    internal First() { }
+    public void SomeMethod() { }
+  }
+}
+",
+                @"
+namespace CompatTests
+{
+  public class First
+  {
+    protected First() { }
+    public void SomeMethod() { }
+    public abstract void SomeAbstractMember() { }
+  }
+}
+"
+            };
+        }
+
+        public static IEnumerable<object[]> AddedAbstractMemberIsReportedData()
+        {
+            yield return new object[]
+            {
+                @"
+namespace CompatTests
+{
+  public class First
+  {
+    public abstract void FirstAbstract() { }
+  }
+}
+",
+                @"
+namespace CompatTests
+{
+  public class First
+  {
+    public abstract void FirstAbstract() { }
+    public abstract string SecondAbstract() => throw null;
```

Comment:
Are these meant to be syntactically correct?  https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANmgExAGoAfAOwEMBbGAZwAdKwYACAYQmqYwBV6MdALAAoAN6jWrAAIBmGQCZWAMWxQ6GSawkipUua0rANUZhhkoVajQEFjGU2AwAKAJTbWAXy375Rk2YyAIwADKwAyjCQ5AR2AU5urAC8AHysGAAWUBAA7qzkAK64uADcWt4inkA

Reply:
Good catch. They shouldn't affect the tests, but we I should fix them.